remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/filter/JPABinaryProtobufFilterAndConverter.java

```diff
@@ -0,0 +1,92 @@
+package org.infinispan.query.remote.impl.filter;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.filter.AbstractKeyValueFilterConverter;
+import org.infinispan.metadata.Metadata;
+import org.infinispan.objectfilter.ObjectFilter;
+import org.infinispan.protostream.ProtobufUtil;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.query.remote.impl.ExternalizerIds;
+import org.infinispan.query.remote.impl.ProtobufMetadataManagerImpl;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Adapter for {@link JPAProtobufFilterAndConverter} that produces binary values as a result of filter/conversion.
+ *
+ * @author gustavonalle
+ * @since 8.1
+ */
+public class JPABinaryProtobufFilterAndConverter<K, V> extends AbstractKeyValueFilterConverter<K, V, Object> {
+
+   private transient SerializationContext serCtx;
+   private final JPAProtobufFilterAndConverter delegate;
+
+   @Inject
+   @SuppressWarnings("unused")
+   protected void injectDependencies(Cache cache) {
+      delegate.injectDependencies(cache);
+      serCtx = ProtobufMetadataManagerImpl.getSerializationContextInternal(cache.getCacheManager());
+   }
+
+   public JPABinaryProtobufFilterAndConverter(String jpaQuery, Map<String, Object> namedParameters) {
+      this.delegate = new JPAProtobufFilterAndConverter(jpaQuery, namedParameters);
+   }
+
+   private JPABinaryProtobufFilterAndConverter(JPAProtobufFilterAndConverter delegate) {
+      this.delegate = delegate;
+   }
+
+   @Override
+   public Object filterAndConvert(K key, V value, Metadata metadata) {
+      Optional<ObjectFilter.FilterResult> filterResult = Optional.ofNullable(delegate.filterAndConvert(key, value, metadata));
```

Comment:
Neither the other java 8 features above  :/
I was assume neither of this will need to be sent back in time, wdyt?